File: factory/python3/StructureBuildImpl.py
import random
import time
from abc import ABCMeta, abstractmethod
from IStructureFactory import IStructureFactory

class StructureBuildImpl(IStructureFactory):
	__materials = ['Material ZX','Material ZF','Material ZT']
    
	def getMaterials(self, object):
		for material in self.__materials:
			print("Need {0} of {1} to start a house.".format(random.randint(1,101), material))
		print("-------------------------------------")

	def initBuilding(self):
		print("This is plan to building...")
		for plan in range(5):
			print("Step {0} in {1} minutes.".format(plan, random.randint(1,101)))
		print("-------------------------------------")

	def stopBuilding(self, object):
		print("")

File: factory/java/src/main/java/co/com/factory/sample/structure/impl/StructureBuildImpl.java
package co.com.factory.sample.structure.impl;

import co.com.factory.sample.constans.Constans;
import co.com.factory.sample.structure.IStructureFactory;
import co.com.factory.sample.utils.GenericUtils;

public class StructureBuildImpl implements IStructureFactory {

    private String[] materials = {"Material ZX","Material ZF","Material ZT"};

    @Override
    public void getMaterials() {
        for (String material : materials) {
            GenericUtils.infoLogger(String.format("Need %s of %s to start a %s.",
                    (int) (Math.random()*100), material, Constans.BUILD));
        }
        GenericUtils.infoLogger("-------------------------------------");
    }

    @Override
    public void initBuilding() {
        GenericUtils.infoLogger("This is plan to building...");
        for (int plan = 1; plan <= 5; plan++){
            GenericUtils.infoLogger(String.format("Step %s in %s minutes.", plan, (int) (Math.random()*100)));
        }
        GenericUtils.infoLogger("-------------------------------------");
    }

    @Override
    public void stopBuilding() {
        GenericUtils.infoLogger("Init stopBuilding in StructureBuildImpl");
    }
}
